src/docfx/lib/process/ProcessUtility.cs

```diff
@@ -73,19 +79,19 @@ static void PipeStream(TextReader input, TextWriter output1, TextWriter? output2
 
     /// <summary>
     /// Reads the content of a file.
-    /// When used together with <see cref="WriteFile(string,string)"/>, provides inter-process synchronized access to the file.
+    /// When used together with <see cref="WriteJsonFile(string,string)"/>, provides inter-process synchronized access to the file.
     /// </summary>
     public static T ReadJsonFile<T>(string path) where T : class, new()
     {
-        var content = "";
+        byte[] bytes;
         using (InterProcessMutex.Create(path))
         {
-            content = File.ReadAllText(path);
+            bytes = File.ReadAllBytes(path);
         }
 
         try
         {
-            return JsonUtility.DeserializeData<T>(content, new FilePath(path));
+            return JsonSerializer.Deserialize<T>(bytes, s_jsonOptions) ?? new();
```

Comment:
Instead of using `JsonSerializer` directly, how about having another central place for `system.text.json`? Like the current `JsonUtility`

Reply:
Yes, I was thinking about consolidating it to `JsonUtility` later, but now that you brought up.